### DISCUSSION TITLE
Plan to support Renovate configuration?

### DISCUSSION THREAD

**Discussion by shinGangan** [honojs/hono#2246] [Ideas]
### Context

Thank you for developing the wonderful framework `Hono`. I appreciate it. 🙏

I usually prepare the following renovate-config in my development to manage the library version.

- [shinGangan/renovate-config/configs/hono.json](https://github.com/shinGangan/renovate-config/blob/d3dd53c9279a0691286ab30dca7017fea2f6cef9/configs/hono.json)

Hono, HonoX, hono middleware, and more Hono families will be added in the future.
Therefore, I think it would be better to provide renovate config on the official side of honojs organization.

### Discuss

If renovate config is provided by the official side, I think it would be preferable to create a renovate-config repository (eg. [nuxt/renovate-config-nuxt](https://github.com/nuxt/renovate-config-nuxt))
(since renovate config is not middleware either).

How to manage this needs to be discussed.

### My opinion.　
　
Middleware is a collection of npm packages used by users. Therefore, I think it is better to have a separate repository for renovate configuration which is not npm packages.

### 📗 Notes

This is not a main feature development, so I consider it a very low priority for the development core team.
Therefore, I would like to be able to contribute to this project as I have already implemented it.

**Comment by yusukebe** [comment]
Hi @shinGangan !

I hadn't really thought much about that. There is a possibility of doing that, but not right away. More repositories would be much more expensive to maintain, even if someone contributes. But I think we can leave the discussion open.

**Reply by shinGangan** [reply]
@yusukebe , Thank you for your comment. 🙏

>  More repositories would be much more expensive to maintain, even if someone contributes.

I agree with you about the increased maintenance costs. I also agree that as a maintainer team, we don't want to increase administrative costs as much as possible.

However, due to the infrequently updated nature of Renovate config, we believe that management costs are not higher than for other middleware.
The biggest factor is the initial installation cost.

#### My Opnion 🙋

I have several other breakthroughs in my mind.

1. [renovate](https://github.com/renovatebot/renovate) presets official additional issues and PRs.
  - pros: official will manage the presets, so future maintenance will be more generous
  - cons: there are currently over 500 issues and it is doubtful when they will be addressed.

2. set up and temporarily manage a repository of [`renovate-config-hono`](https://github.com/shinGangan/renovate-config-hono) on my end.
  - pros: The maintainer team can focus on feature development since it is not under the control of the maintainer team.
  - cons: user recognition and other issues may arise.

I will continue to manage Renovate config (as I use Hono for personal and product use).
Therefore, I think it is fine to manage it in my personal repository until the need for it increases.


I would like to hear @yusukebe 's view.

**Reply by yusukebe** [reply]
To be honest, I prefer the way you are managing it now in your private repository. This is for the same reasons as I have already mentioned.

But you can write that in [hono.dev](https://hono.dev).

You could write in the "Ecosystem" section of the following section, though might be a better place elsewhere.

https://hono.dev/snippets/top

**Reply by shinGangan** [reply]
> To be honest, I prefer the way you are managing it now in your private repository. This is for the same reasons as I have already mentioned.

I also support this opinion. 🙆

<br />

> You could write in the "Ecosystem" section of the following section, though might be a better place elsewhere.

I have one suggestion on this.That is to have a **FAQs** section.
This is inspired by the Panda CSS documentation and is the most helpful documentation I have seen so far.

- ref: https://panda-css.com/docs/overview/faq

As Hono develops and is used more and more, we can expect to see more and more "non-bug reports" and "rudimentary questions that have been answered many times" being voted into issues.
Therefore, by preparing a FAQs item, it will be possible to guide people to "check the FAQs, and if there is no similar item, ask a question in GitHub Discussion (or Issue)", which will reduce the management cost for the maintainer team.

**Reply by yusukebe** [reply]
"**FAQs** section" sounds great! I'd like to have that!

If we don't have a lot of content, we could create a page in `/guides/faq`. I think that would be fine at first. If you can do that, contributions are welcome.

**Reply by shinGangan** [reply]
I am glad you liked it and I suggested it!
I will take care of creating the `/guides/faq` page 🙆

**Reply by shinGangan** [reply]
Issue to be tied together and corresponding PR created.

- honojs/website#251
- honojs/website#252

**Comment by shinGangan** [comment]
Separate thread from the above discussion as it is a separate agenda item.

### Q. Do you want to add a Renovate setup for the development team as well?

When we provide Renovate configuration to users, we think it would be good to be able to semi-automate the update of libraries used during development.
This would allow the development team to take advantage of Renovate's strengths.

### Points to be solved

In the operation of Renovate, we would like to discuss

- Who will do the reviews?
- Frequency of PR creation

We do not want to increase operational costs due to tools that facilitate development team operations.

### Experiment

We are currently experimenting with HonoX.

shinGangan/renovate-config-hono#9

**Reply by shinGangan** [reply]
@yusukebe , I would like to hear @yusukebe 's view. 🙇

**Reply by yusukebe** [reply]
Does this mean you recommend that we use Renovate in our Hono project?

I was thinking of introducing Renovate or Dependabot in `honojs/starter`, first.

**Reply by shinGangan** [reply]
`Dependabot`,`Renovate`, I guess it is the development team's decision which one to choose.
I think Renovate is better for the following two reasons

1. it can inherit configuration files created in the past.

Renovate has an `extends` option. It is possible to inherit and use `renovate-config-hono` created by me. 
It is also possible to cut out and inherit common settings used in all repositories to a separate repository (eg. common-linters, common-config-test, etc.).

2. Presets provided by official vendors can be used.

Renovate offers a number of presets provided by official vendors.
Settings used by many Renovate users, such as JEST and LINTER, are already provided as presets.
    
  - ref: https://docs.renovatebot.com/presets-default/

Whichever decision is made, I support the development team's decision.
